systemtest/src/test/java/io/strimzi/systemtest/upgrade/OlmUpgradeIsolatedST.java

```diff
@@ -55,117 +54,121 @@ public class OlmUpgradeIsolatedST extends AbstractUpgradeST {
-        LOGGER.info("Deploy Kafka from: {}", kafkaYaml.getPath());
+        LOGGER.info("Deploy Kafka from file: {}", kafkaYaml.getPath());
         KubeClusterResource.cmdKubeClient().create(kafkaYaml);
-        // Wait for readiness
         waitForReadinessOfKafkaCluster();
 
-        this.kafkaUpgradeTopic = new YAMLMapper().readValue(new File(dir, upgradeData.getFromExamples() + "/examples/topic/kafka-topic.yaml"), KafkaTopic.class);
-        this.kafkaUpgradeTopic.getMetadata().setName(topicUpgradeName);
-        this.kafkaUpgradeTopic.getSpec().setReplicas(3);
-        this.kafkaUpgradeTopic.getSpec().getConfig().put("min.insync.replicas", 2);
+        // Create KafkaTopic
+        final String topicUpgradeName = "topic-upgrade";
+        HashMap<String, Object> topicConfig = new HashMap<String, Object>();
+        topicConfig.put("min.insync.replicas", 2);
+
+        KafkaTopic kafkaUpgradeTopic = new YAMLMapper().readValue(new File(dir, upgradeData.getFromExamples() + "/examples/topic/kafka-topic.yaml"), KafkaTopic.class);
+        kafkaUpgradeTopic = new KafkaTopicBuilder(kafkaUpgradeTopic)
+            .editMetadata()
+            .withNamespace(clusterOperator.getDeploymentNamespace())
+                .withName(topicUpgradeName)
+            .endMetadata()
+            .editSpec()
+                .withReplicas(3)
+                .addToConfig(topicConfig)
+            .endSpec()
+            .build();
 
-        LOGGER.info("Deploy KafkaTopic: {}", this.kafkaUpgradeTopic.toString());
+        resourceManager.createResource(extensionContext, kafkaUpgradeTopic);
 
-        cmdKubeClient().applyContent(this.kafkaUpgradeTopic.toString());
-        ResourceManager.waitForResourceReadiness(KafkaTopic.RESOURCE_PLURAL + "." +
-            io.strimzi.api.kafka.model.Constants.V1BETA2 + "." + io.strimzi.api.kafka.model.Constants.RESOURCE_GROUP_NAME, topicUpgradeName);
+        KafkaClients kafkaBasicClientJob = new KafkaClientsBuilder()
+            .withProducerName(testStorage.getProducerName())
+            .withConsumerName(testStorage.getConsumerName())
+            .withNamespaceName(testStorage.getNamespaceName())
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(clusterName))
+            .withTopicName(topicUpgradeName)
+            .withMessageCount(testStorage.getMessageCount())
+            .withDelayMs(1000)
+            .build();
 
-        resourceManager.createResource(extensionContext, kafkaBasicClientJob.producerStrimzi());
-        resourceManager.createResource(extensionContext, kafkaBasicClientJob.consumerStrimzi());
+        resourceManager.createResource(extensionContext, kafkaBasicClientJob.producerStrimzi(), kafkaBasicClientJob.consumerStrimzi());
+        ClientUtils.waitForClientsSuccess(testStorage.getProducerName(), testStorage.getConsumerName(), clusterOperator.getDeploymentNamespace(), testStorage.getMessageCount());
```

Comment:
You should just create the clients and then wait for success at the end of the test. We are verifying that during the upgrade we are able to produce and consume messages.